[BREAKING] Process coding that can be influenced by user-input without `-fp` - speeds up playing with tags

treee111 opened the pull request:
## This PR…

- Enhances function `filter_tags_from_country_osm_pbf_files` to speed up processing when playing with filtering tags / tag-mapping / tag-transform

## Considerations and implementations

Coding that can be influeced by user-input:
- Filtering unwanted map objects out of maps
- Splitting filtered country files to tiles
- Merging splitted tiles with land and sea
- Creating map tiles - apply tag-wahoo xml

Coding that can not be influeced by user-input:
- Converting map to o5m format (Windows only)

## How to test

checkout this branch or install version v2.1.0a14 via `pip install wahoo_mc==2.1.0a14`

1. Run `python -m wahoomc cli -co malta -tag tag-wahoo.xml -fp`
2. check size and content of generated maps
3. `python -m wahoomc cli -co malta -tag tag-wahoo-poi.xml`
4. check size and content of generated maps

The files from 2. and 4. should be different. This would show, that the both .xml files have done their job.

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md)
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/.github/CONTRIBUTING.md#Pull-Requests) section
- [x] Commits (and commit-messages) are understandable
- [x] Tested with macOS / Linux
- [x] Tested with Windows

zenziwerken committed:
Update osm_maps_functions.py

According to the documentation 
https://wiki.openstreetmap.org/wiki/Osmfilter#Keep_only_specific_Tags
tags have to be enclosed with >"<

zenziwerken committed:
Update osm_maps_functions.py

Keep out_file_o5m if not force_processing to speed up build process when updating filtering tags / tag-mapping / tag-transform

treee111 committed:
finalization of @zenziwerken 's work

pylint finding, comment

treee111 committed:
catch error if file does not exist but should be deleted

e.g. if processing -fp after switching from v2.0.2 to this version (Windows only)

treee111 commented:
Hi @zenziwerken,
as announced I made a separate PR for the changes for potential speed gains.
I tested it like written in the "How to test" section. The generated files had no differences, I needed a -fp on the second command to get the poi xml interpreted and other output. This is not what the achievement should be if I understood it correc.

Are the changes in this PR like you proposed?
Is description and the "How to test" written by me the same as you thought with these changes?

Thanks!

zenziwerken commented:
I did not make myself sufficiently clear. The result should be the same. It just skip converting the original pfb-file from OSM to o5m every time when playing with filtering tags / tag-mapping / tag-transform.

treee111 commented:
> I did not make myself sufficiently clear. The result should be the same. It just skip converting the original pfb-file from OSM to o5m every time when playing with filtering tags / tag-mapping / tag-transform.

This was also what I interpreted from your changes, but thought I was not having the real understanding.

I think I got now what would give a speed gain:
1. skip the pfb-file to o5m file creation if the o5m file is there already
why: because there is no user-influence possible
2. do o5m file creation when `-fp` is given or o5m file does not exist
3. do the filtering with constants every time
why: because there is a possible user-influence
4. apply the tag-wahoo .xml file during map creation every time
why: because there is a possible user-influence

After that, `-fp` has kind of changes the character in these situations and we can further check if the `-fp` is still valid, can be deleted or can be combined with `-fd`

zenziwerken commented:
Perfect. That's it!

treee111 committed:
adjust constants processing based on discussion in PR

https://github.com/treee111/wahooMapsCreator/pull/150#issuecomment-1261475761

treee111 committed:
adjust tag-wahoo.xml processing based on discussion in PR

https://github.com/treee111/wahooMapsCreator/pull/150#issuecomment-1261475761

treee111 committed:
adjust splitting to tiles processing based on discussion in PR

https://github.com/treee111/wahooMapsCreator/pull/150#issuecomment-1261475761

treee111 committed:
adjust merging with land and sea processing based on discussion in PR

https://github.com/treee111/wahooMapsCreator/pull/150#issuecomment-1261475761

treee111 committed:
[FEAT] Refactor a lot to decouple and make maintenance easier (focus osm_maps_functions.py) (#152)

* import file_directory functions one by one

- and not by importing the whole file "as" fd_fct

* create OsmData object and insert into class OsmMaps

* move check for -co and -xy to input.py

* shuffle a lot

* go throught calc_border_countries also when no border countries are wanted

* break OsmData in small pieces

* refactor border country calc + fix unittests

* force_processing refactoring, all checks in OsmData now!

- move to function process_input_of_the_tool
- move corresponding stuff from check_and_download_files also
- additional needed changes - mainiy in downloader.py
- adjust tests & do nearly refactoring

* import needed functions + pylint findings

* refactor imports to "single import" and not the whole file

* fix pylint finding

* Bump to version v2.1.0a13

treee111 committed:
fix pylint finding

treee111 committed:
harmonize comments

treee111 commented:
Hi @zenziwerken,
so I am done now in my opinion. You can have a look at the changes in the "Files changed" tab. "Hide whitespaces" makes the diff not so hugh.
Could you please test in your Windows environment if everything works as desired?
I created v2.1.0a14 for easy test-install.
Thanks! Greetings

treee111 committed:
Bump to version v2.1.0a14

zenziwerken commented:
Everything looks perfect. Mapcreation runs smoothly under Windows.

treee111 commented:
> Everything looks perfect. Mapcreation runs smoothly under Windows.

thanks!

treee111 committed:
Revert to version v2.0.2
